library/core/src/array/mod.rs

```diff
@@ -427,3 +427,78 @@ impl<T, const N: usize> [T; N] {
         unsafe { crate::mem::transmute_copy::<_, [U; N]>(&dst) }
     }
 }
+
+/// Creates an array `[T; N]` where each array element `T` is returned by the `cb` call.
+///
+/// # Arguments
+///
+/// * `cb`: Callback where the passed argument is the current array index.
+///
+/// # Example
+///
+/// ```rust
+/// let array: [usize; 5] = core::array::create_array(|i| i);
+/// assert_eq!(array, [0, 1, 2, 3, 4]);
+/// ```
+#[inline]
+#[unstable(feature = "create_array", issue = "0")]
+pub fn create_array<F, T, const N: usize>(mut cb: F) -> [T; N]
+where
+  F: FnMut(usize) -> T,
+{
+    let mut array: MaybeUninit<[T; N]> = MaybeUninit::uninit();
+    // SAFETY: It is guaranteed that the whole array will be filled
+    unsafe {
+        for (idx, value_ptr) in (&mut *array.as_mut_ptr()).iter_mut().enumerate() {
+            ptr::write(value_ptr, cb(idx));
+        }
+        array.assume_init()
+    }
+}
+
+/// Creates an array `[T; N]` where each fallible array element `T` is returned by the `cb` call.
+/// Unlike `create_array`, where the element creation can't fail, this version will return an error
+/// if any element creation was unsuccessful.
+///
+/// # Arguments
+///
+/// * `cb`: Callback where the passed argument is the current array index.
+///
+/// # Example
+///
+/// ```rust
+/// use core::array::create_array_rslt;
+///
+/// #[derive(Debug, PartialEq)]
+/// enum SomeError {
+///     Foo
+/// }
+///
+/// let array: Result<[usize; 5], SomeError> = create_array_rslt(|i| Ok(i));
+/// assert_eq!(array, Ok([0, 1, 2, 3, 4]));
+///
+/// let another_array: Result<[usize; 5], SomeError> = create_array_rslt(|_| Err(SomeError::Foo));
+/// assert_eq!(array, Err(SomeError::Foo));
+/// ```
+#[inline]
+#[unstable(feature = "create_array", issue = "0")]
+pub fn create_array_rslt<E, F, T, const N: usize>(mut cb: F) -> Result<[T; N], E>
```

Comment:
```suggestion
pub fn try_create_array<E, F, T, const N: usize>(mut cb: F) -> Result<[T; N], E>
```

Reply:
Most people voted for `from_fn` and `try_from_fn`. Marking as resolved.